src/components/atoms/Button/Button.tsx

```diff
@@ -0,0 +1,60 @@
+// @ts-nocheck
+import React from "react";
+
+// Typings
+import { ButtonProps } from "./Button.types";
+
+// Styles
+import * as S from "./Button.styles";
+import { Link } from "react-router-dom";
+
+const AppButton: React.FC<ButtonProps> = ({
+  customClass,
+  loading,
+  onClick,
+  text,
+  type,
+  customStyle,
+  children,
+  gradient,
+  isLink = false,
+  linkTo,
+}) => {
+  if (loading)
+    return (
+      <div className="spinner-border">
+        <span className="sr-only">Loading...</span>
+      </div>
+    );
+
+  if (isLink) {
+    return (
+      <S.Button
+        as={Link}
+        className={customClass}
+        style={customStyle}
+        to={linkTo!}
+      >
+        {children ?? text}
+      </S.Button>
+    );
+  }
+
+  return (
+    <S.Button
+      className={customClass}
+      style={customStyle}
+      type={type}
+      onClick={onClick}
+      hasGradient={gradient}
+    >
+      {children ?? text}
+    </S.Button>
+  );
+};
+
+AppButton.defaultProps = {
+  type: "button",
```

Comment:
[NB] I think we should stick to one way of setting defultProps

Reply:
Agreed that we should only have one way. From memory there were a few advantages that can be gained by doing it this way.. but i'm not sure we really care about them. Unless there is good evidence as to why we should do it differently, my personal preference is to use standard ES6+ syntax